File: qe-selenium-java/Activity4.java
package demos;

import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class Activity4 {
public static void main(String[] args) {
    WebDriver driver = new FirefoxDriver();
    driver.get("https://training-support.net/webelements/target-practice");
    System.out.println(driver.getTitle());

    WebElement header3 = driver.findElement(By.xpath("//h3[text() = 'Heading #3']"));
    // WebElement header3 = driver.findElement(By.cssSelector("h3.text-3xl:nth-child(5)"));

    WebElement header5 = driver.findElement(By.xpath("//h5[text() = 'Heading #5']"));
      WebElement purpleButton = driver.findElement(By.cssSelector("button.rounded-xl:nth-child(9)"));
      

    // WebElement element3 = driver.findElement(By.cssSelector("button.rounded-xl:nth-child(9)"));
    WebElement slateButton = driver.findElement(By.xpath("//button[text()='Slate']"));


    // String text = header3.getText();
    // String text = header5.getCssValue("color");
    System.out.println("Header3: "+header3.getText());
    System.out.println("Header5 color: "+header5.getCssValue("color"));

    String purpleButtonClasses = purpleButton.getAttribute("class");
    System.out.println("Purple Button All Classes: "+purpleButtonClasses);

    String slateButtonText = slateButton.getText();
    System.out.println("Slate Button Text: "+slateButtonText);

    driver.quit();
    
}
}

File: qe-selenium-python/Activity4.py
from selenium import webdriver
from selenium.webdriver.common.by import By

with webdriver.Firefox() as driver:
    driver.get('https://training-support.net/webelements/target-practice')
    print(driver.title)

    header3 = driver.find_element(By.XPATH, '//h3[text() = "Heading #3"]')
    header5 = driver.find_element(By.XPATH, '//h5[text() = "Heading #5"]')

    print("Header3 Text: ", header3.text)
    print("Header5 Color: ", header5.text)

    purple_btn = driver.find_element(By.CSS_SELECTOR, 'button.rounded-xl:nth-child(9)')
    slate_btn = driver.find_element(By.XPATH, "//button[text() = 'Slate']")

    print("Purle Button All Classes: ", purple_btn.get_attribute('class'))
    print("Slate Button Text: ", slate_btn.text)

    driver.quit()